scene/main/node.cpp

```diff
@@ -2422,7 +2422,9 @@ void Node::_replace_connections_target(Node *p_new_target) {
 
 		if (c.flags & CONNECT_PERSIST) {
 			c.source->disconnect(c.signal, this, c.method);
-			ERR_CONTINUE(!p_new_target->has_method(c.method));
+			bool invalid = !p_new_target->has_method(c.method) && (!p_new_target->get_script().is_null() && !Ref<Script>(p_new_target->get_script())->has_method(c.method));
```

Comment:
You should reverse the logic to test if it's `valid` instead, then do `ERR_CONTINUE(!valid)`. It'll make the condition easier to understand.

Make sure to amend the commit after the change. You can consult our [PR workflow guide](http://docs.godotengine.org/en/latest/community/contributing/pr_workflow.html#modifying-a-pull-request) for instructions.